src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/containerEnvironmentIsHonored.groovy

```diff
@@ -0,0 +1,28 @@
+podTemplate(yaml: '''
+spec:
+  containers:
+  - name: jnlp
+    image: jenkins/inbound-agent:3192.v713e3b_039fb_e-1
+    # PATH=/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+  - name: alpine
+    image: alpine:3.19.0
+    # PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+    command:
+    - sleep
+    args:
+    - infinity
+''') {
+  node(POD_LABEL) {
+    echo "from Groovy: ${env.PATH}"
```

Comment:
Also interesting to check this at various points. Your proposed fix using Bourne shell expansion (`eval`?) in the `ContainerExecDecorator` might fix the `inside container with override` from `sh` but would not fix it from Groovy. Whether anyone _cares_ is another question, it just might be a source of confusion if the Groovy view of the environment differs from what `sh` steps actually use.

Reply:
Pure groovy ought to be always executed on the controller, independently from whether we are within a node or a container, so to be this is not a problem.